electron/main.ts

```diff
@@ -348,23 +362,29 @@ ipcMain.handle('install', async (event, args) => {
     command = `${legendaryBin} install ${game} --base-path ${defaultInstallPath} ${workers} -y |& tee ${logPath}`
   }
   console.log(`Installing ${game} with:`, command)
-  const noSpaceMsg = 'Not enough available disk space'
-  return await execAsync(command, { shell: '/bin/bash' })
-    .then(() => console.log('finished installing'))
-    .catch(async () => {
-      try {
-        const { stdout } = await execAsync(
-          `tail ${logPath} | grep 'disk space'`
-        )
+  try {
+    await execAsync(command, { shell: '/bin/bash' })
+    console.log('finished installing')
+  } catch (error) {
+    const noSpaceMsg = 'Not enough available disk space'
+    const genericMessage = 'installation canceled or had some error'
+    await execAsync(`tail ${logPath} | grep 'disk space'`)
+      .then(({ stdout }) => {
         if (stdout.includes(noSpaceMsg)) {
           console.log(noSpaceMsg)
-          return stdout
+          return showErrorBox(
```

Comment:
I see this function returns either a string or void (showErrorBox) where the returned value does nothing. Maybe its good to just remove the returns if it isn't being used in the end?

Reply:
Good catch. I'm refactoring it to be more generic as well.